src/plays/registration-form/RegistrationForm.jsx

```diff
@@ -0,0 +1,238 @@
+      if (!regx.test(emailvalue[1])) {
+        //set error and return false
+        setValues((prevState) => ({
+          ...prevState,
+          errors: {
+            ...prevState.errors,
+            email: `your email is not correct`,
+          },
+        }));
+        return false;
+      } else {
+        //if email is correct clear error msg and return true
+        setValues((prevState) => ({
+          ...prevState,
+          errors: {
+            ...prevState.errors,
+            email: ``,
+          },
+        }));
+        return true;
+      }
+    }
+    //if other than email just return true
+    return true;
+  };
+
+  //self explanatory
+  const conmparePassword = (pass, confirmPass) => {
+    // (pass and confirmPass) arrays containing name, value pair eg. ["password","xyz"] and ["confirmpassword","xyz"]
+    if (pass[0] === "password") {
+      if (values.confirmPassword !== "") {
+        if (pass[1] !== confirmPass[1]) {
+          //values don't match set error, return false
+          setValues((prevState) => ({
+            ...prevState,
+            errors: {
+              ...prevState.errors,
+              password: "Password doesn't match",
+              confirmPassword: "Password doesn't match",
+            },
+          }));
+          return false;
+        } else {
+          //if matches remove error msgs, return true
+          setValues((prevState) => ({
+            ...prevState,
+            errors: {
+              ...prevState.errors,
+              password: "",
+              confirmPassword: "",
+            },
+          }));
+          return true;
+        }
+      }
+      //return true in any other condition
+      return true;
+    }
+    //return true in any other condition
+    return true;
+  };
+
+  //final validation
+  const validate = (value) => {
+    //value is an array ex. ["name","deepak"]
+    if (value[1] === "") {
+      //if empty set error msg return false
+      setValues((prevState) => ({
+        ...prevState,
+        errors: {
+          ...prevState.errors,
+          [value[0]]: `Please fill the ${value[0]}`,
+        },
+      }));
+      return false;
+    } else {
+      //if not empty remove message return (validate email && compare password)
+      setValues((prevState) => ({
+        ...prevState,
+        errors: {
+          ...prevState.errors,
+          [value[0]]: ``,
+        },
+      }));
+      return (
+        validateEmail(value) &&
+        conmparePassword(value, Object.entries(values)[3])
+      );
+    }
+  };
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+    let counter = 0;
+    Object.entries(values).forEach((value, i) => {
+      //validate each field, increase counter if validation passed.
+      if (validate(value)) {
+        counter = counter + 1;
+      }
+    });
+
+    //if all fields are validated succsfully store the values in new object and clear the input values
+    if (counter === 5) {
```

Comment:
It may not be the best way to handle in a bigger scale application. So everytime there is a new field you need to change this condition to match with a new number. Can you think of a better way?

Note: I can accept with the current code but I think, it is ok to put more thoughts into it, what say?